### PR TITLE
[Fix] MotionBlur bug fix and doctest update

### LINKED ISSUE #779: [Bug] 2D motion blur kernel seems wrong
## 🐛 Bug

I'm seeing some issues with the kernel returned by the call `kornia.filters.get_motion_kernel2d()`

## To Reproduce

1. kornia.filters.get_motion_kernel2d(5, 0., 0.)
2. kornia.filters.get_motion_kernel2d(3, 215., -0.5)

## What I am seeing
```sh
>>> kornia.filters.get_motion_kernel2d(5, 0., 0.)
tensor([[[nan, nan, nan, nan, nan],
         [nan, nan, nan, nan, nan],
         [inf, inf, nan, -inf, -inf],
         [nan, nan, nan, nan, nan],
         [nan, nan, nan, nan, nan]]])
>>> kornia.filters.get_motion_kernel2d(3, 215., -0.5)
tensor([[[-0.0000, -0.0000, -0.3333],
         [-0.0000,  0.3333, -0.0000],
         [ 1.0000, -0.0000, -0.0000]]])
```

## Expected behavior

According to https://github.com/kornia/kornia/blob/master/kornia/filters/kernels.py#L634, I would expect 
```sh
>>> kornia.filters.get_motion_kernel2d(5, 0., 0.)
tensor([[0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
              [0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
              [0.2000, 0.2000, 0.2000, 0.2000, 0.2000],
              [0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
              [0.0000, 0.0000, 0.0000, 0.0000, 0.0000]])
>>> kornia.filters.get_motion_kernel2d(3, 215., -0.5)
tensor([[0.0000, 0.0412, 0.0732],
              [0.1920, 0.3194, 0.0804],
              [0.2195, 0.0743, 0.0000]])
```

## Environment

Collecting environment information...
PyTorch version: 1.7.0
Is debug build: True
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Mac OSX 10.15.7 (x86_64)
GCC version: Could not collect
Clang version: 12.0.0 (clang-1200.0.32.2)
CMake version: Could not collect

Python version: 3.7 (64-bit runtime)
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.16.1
[pip3] torch==1.7.0
[pip3] torchvision==0.8.1
[conda] Could not collect

## Extra info

I am seeing this issue on both mac and linux, both with pytorch 1.7 and kornia 0.4.1

### PR DESCRIPTION
**Pull request by shijianjian**
### Description
A few sentences describing the changes proposed in this pull request.

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

**Review by edgarriba — COMMENTED** [review]
Can you add tests ? I would add also the module to run in the doctest routine

**Comment by shijianjian** [comment]
The doctest for this was implemented in the first place. We need to add the doctest into the CI.

**Comment by edgarriba** [comment]
It's running already. Just need to add stuff incremental ly https://github.com/kornia/kornia/blob/master/Makefile#L41

**Review by edgarriba — CHANGES_REQUESTED** [review]
add some tests checking the input/output shapes and numerics. Update make doctests with including filters (or maybe in another PR)

**Comment by shijianjian** [comment]
@edgarriba Please merge this one. It is the fix for a bit serious bug that could output nan values.

File: kornia/filters/kernels.py [hunk 1]
```diff
@@ -622,21 +622,20 @@ def get_motion_kernel2d(kernel_size: int, angle: Union[torch.Tensor, float],
...
-                [0.2195, 0.0743, 0.0000]])
+        - Output: :math:`(B, ksize, ksize)`
+
+    Examples::
+        >>> get_motion_kernel2d(5, 0., 0.)
+        tensor([[[0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
+                 [0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
+                 [0.2000, 0.2000, 0.2000, 0.2000, 0.2000],
+                 [0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
+                 [0.0000, 0.0000, 0.0000, 0.0000, 0.0000]]])
```
Comment (edgarriba) [inline kornia/filters/kernels.py:633]: @shijianjian what's the lògic for angles here ? I would expect this 90deg. Probably documento need to specify this too.
Is there to compare against ?

File: kornia/filters/kernels.py [hunk 2]
```diff
@@ -698,34 +698,52 @@ def get_motion_kernel3d(kernel_size: int, angle: Union[torch.Tensor, Tuple[float
         torch.Tensor: the motion blur kernel.
 
     Shape:
-        - Output: :math:`(ksize, ksize)`
+        - Output: :math:`(B, kernel_size, kernel_size, kernel_size)`
```
Comment (edgarriba) [inline kornia/filters/kernels.py:701]: For next PRs. The convention I try to follow for functionals is to add the input/output shapes in the arts or return itself. Faster to read.

**Review by edgarriba — APPROVED** [review]
LGTM

File: kornia/filters/kernels.py (hunk 1 above)
Comment (shijianjian) [reply kornia/filters/kernels.py:633]: I did not get it. This kernel is firstly determined by the direction, the rotate by the specific angles as explained in the doc. Do you mean to -90 degrees to let angle ==0 as vertical?

Comment (edgarriba) [reply kornia/filters/kernels.py:633]: yes

**Pull request merged**